### ISSUE TITLE
Implement key id validation in Step 5

### ISSUE THREAD

**Issue by valeneiko** [srinivas1729/appattest-checker-node#1]
Step 5 here is a TODO:
https://github.com/srinivas1729/appattest-checker-node/blob/581b5b5140b090ab290b7d84fbb63019040f018a/src/attestation.ts#L202-L217

The docs say we need to validate sha256 hash of the public key. In reality this means a hash of a 65-byte string `0x04 ${key.x} ${key.y}`. You can get x and y value by encoding the public key in `jwk` format, or get the complete string you need to hash from the last 65 bytes of the der encoded public key.

So in NodeJS you would do something like this:
```js
const credCert = new X509Certificate(credCertBufferFromAttestationObject);
const publicKey = credCert.publicKey;
const keyBuf = publicKey.export({format: 'der', type: 'spki'});
const publicKeyParam = keyBuf.subarray(-65);
const keyId = createHash('sha256').update(publicKeyParam);
```

or
```js
const credCert = new X509Certificate(credCertBufferFromAttestationObject);
const publicKey = credCert.publicKey;
const keyJwk = publicKey.export({format: 'jwk', type: 'spki'});
const publicKeyParam = Buffer.from([
  0x04,
  ...Buffer.from(keyJwk.x, 'base64'),
  ...Buffer.from(keyJwk.y, 'base64')
]);
const keyId = createHash('sha256').update(publicKeyParam);
```

**Comment by srinivas1729** [comment]
Thanks a lot!  I will incorporate your suggestions in a few days.

**Comment by srinivas1729** [comment]
TODO has been implemented and [change](https://github.com/srinivas1729/appattest-checker-node/commit/03093e2180741b51f89b81b33816e584d838eeb8) was pushed to the repo. I've also released v1.0.1 that includes this fix. Thanks a lot for the guidance!
